src/main/java/com/spotify/annoy/jni/base/AnnoyIndexImpl.java

```diff
@@ -30,53 +30,62 @@
 class AnnoyIndexImpl implements AnnoyIndex {
 
   private final int dim;
+  // Stores the memory location of the tree in cpp that will be passed in
+  // This is how we share states between java and cpp
+  private final long cppPtr;
 
   public List<Integer> getNearestByVector(List<Float> vector, int nbNeighbors) {
     validateVecSize(vector);
-    return primitiveToBoxed(cppGetNearestByVector(boxedToPrimitive(vector), nbNeighbors));
+    return primitiveToBoxed(
+        cppGetNearestByVector(this.cppPtr, boxedToPrimitive(vector), nbNeighbors));
```

Comment:
was just following the how you guys handled the `dim` field. But yeah I do prefer it without the `this`